docs/root/start/sandboxes/tls-inspector.rst

```diff
@@ -0,0 +1,94 @@
+
+TLS Inspector Listener Filter
+=============================
+
+.. sidebar:: Requirements
+
+   .. include:: _include/docker-env-setup-link.rst
+
+   :ref:`curl <start_sandboxes_setup_curl>`
+	Used to make ``HTTP`` requests.
+
+   :ref:`jq <start_sandboxes_setup_jq>`
+	Parse ``json`` output from the upstream echo servers.
+
+This example demonstrates how TLS inspector is used to select FilteChain to distribute the traffic
+to different clusters via transport_protocol and application_protocols. And it will show statistics
+of TLS inspector listener filter.
+
+Step 1: Build the sandbox
+*************************
+
+Change directory to ``examples/tls-inspector`` in the Envoy repository.
+
+This starts one proxy listening on ``localhost:10000``.
+
+It also starts three upstream services, one ``HTTP1.1`` and one ``HTTP2`` over ``TLS`` and another
+pure ``HTTP``, which echo back received headers in ``json`` format.
+
+The upstream services listen on the internal Docker network on ports ``443`` and ``80`` respectively.
+
+.. code-block:: console
+
+  $ pwd
+  envoy/examples/tls-inspector
+  $ docker-compose pull
+  $ docker-compose up --build -d
+  $ docker-compose ps
+
+                Name                               Command               State                         Ports
+  ---------------------------------------------------------------------------------------------------------------------------------
+  tls-inspector_service-http_1            docker-entrypoint.sh node  ... Up
+  tls-inspector_service-https-http1.1_1   docker-entrypoint.sh node  ... Up
+  tls-inspector_service-https-http2_1     docker-entrypoint.sh node  ... Up
+  tls-inspector_tls-inspector_1           /docker-entrypoint.sh /usr ... Up      0.0.0.0:10000->10000/tcp, 0.0.0.0:12345->12345/tcp
+
+
+Step 2: Access services
+***********************
+
+Querying the service at port ``10000`` with different ``HTTP`` version specified over ``TLS``, or
+with pure ``HTTP`` protocol. your could see the requests are handled by different services.
+
+.. code-block:: console
+
+   $ curl -sk --http1.1 https://localhost:10000  | jq  '.os.hostname'
+   "service-https-http1.1"
+
+The upstream ``service-https-http1.1`` handles the request.
+
+.. code-block:: console
+
+   $ curl -sk --http2  https://localhost:10000  | jq  '.os.hostname'
+   "service-https-http2"
+
+The upstream ``service-https-http2`` handles the request.
+
+.. code-block:: console
+
+   $ curl -sk http://localhost:10000  | jq  '.os.hostname'
+   "service-http"
+
+The upstream ``service-http`` handles the request. Since TLS Inspector listener filter detects the
+transport is plaintext, it will not set transport_protocol to tls.
+
+Step 3: Get statistics
+**********************
+
+TLS inspector has a statistics tree rooted at tls_inspector, which can be extracted with admin
+access entrypoint configured.
+
+.. code-block:: console
+
+   $ curl -sk http://localhost:12345/stats |grep tls_inspector
```

Comment:
i think this should show the ~expected state after working through the sandbox

after following the docs this is how it showed for me:

```
tls_inspector.alpn_found: 2
tls_inspector.alpn_not_found: 0
tls_inspector.client_hello_too_large: 0
tls_inspector.connection_closed: 0
tls_inspector.read_error: 0
tls_inspector.sni_found: 2
tls_inspector.sni_not_found: 0
tls_inspector.tls_found: 2
tls_inspector.tls_not_found: 1

```

Reply:
Got it.